package/files/nginx-pubcloud/nginx-registry-https.conf

```diff
@@ -0,0 +1,59 @@
+upstream registry {
+    server 127.0.0.1:5000;
+}
+
+## Set a variable to help us decide if we need to add the
+## 'Docker-Distribution-Api-Version' header.
+## The registry always sets this header.
+## In the case of nginx performing auth, the header is unset
+## since nginx is auth-ing before proxying.
+map $upstream_http_docker_distribution_api_version $docker_distribution_api_version {
+  '' 'registry/2.0';
+}
+
+
+server {
+    listen 443   ssl;
+    listen [::]:443 ssl;
+    server_name ~^registry;
+
+    access_log  /var/log/nginx/registry_https_access.log;
+    error_log   /var/log/nginx/registry_https_error.log;
+    root        /var/lib/docker-registry;
+
+    ssl_certificate     /etc/rmt/ssl/susecloud-registry-ec2.crt;
+    ssl_certificate_key /etc/rmt/ssl/susecloud-registry-ec2.key;
+
+    ssl_protocols       TLSv1.2 TLSv1.3;
+
+    # disable any limits to avoid HTTP 413 for large image uploads
+    client_max_body_size 0;
+
+    # required to avoid HTTP 411: see Issue #1486 (https://github.com/moby/moby/issues/1486)
+    chunked_transfer_encoding on;
+
+    location /v2/ {
+      # Do not allow connections from docker 1.5 and earlier
+      # docker pre-1.6.0 did not properly set the user agent on ping, catch "Go *" user agents
+      if ($http_user_agent ~ "^(docker\/1\.(3|4|5(?!\.[0-9]-dev))|Go ).*$" ) {
+        return 404;
+      }
+
+      ## If $docker_distribution_api_version is empty, the header is not added.
+      ## See the map directive above where this variable is defined.
+      add_header 'Docker-Distribution-Api-Version' $docker_distribution_api_version always;
+
+      proxy_pass                          http://registry;
+      proxy_set_header  Host              $http_host;   # required for docker client's sake
+      proxy_set_header  X-Real-IP         $remote_addr; # pass on real client's IP
+      proxy_set_header  X-Forwarded-For   $proxy_add_x_forwarded_for;
+      proxy_set_header  X-Forwarded-Proto $scheme;
+      proxy_read_timeout                  900;
+    }
+
+    # override registry search
+    location /v2/_catalog {
+      # redirect to the catalog endpoint
+      rewrite /v2/_catalog https://smt-ec2.susecloud.net/v2/_catalog redirect;
```

Comment:
So, I woke up overnight and realized - this line is why we'll need to do the one-config approach. We don't want every RMT server in the wild redirecting to `smt-ec2.susecloud.net`... and there's no way to do the redirect without an FQDN.

Reply:
yes, hard coded string had to be fixed